src/DMXSerial2.cpp

```diff
@@ -341,6 +342,13 @@ void DMXSerialClass2::init(struct RDMINIT *initData, RDMCallbackFunction func, R
 
   _baseInit();
 
+  // Sanity check/fix the defaultPersonalityNumber
+  if ((_initData->defaultPersonalityNumber < DMXSERIAL_MIN_PERSONALITY_VALUE) || (_initData->defaultPersonalityNumber > min(_initData->personalityCount, DMXSERIAL_MAX_PERSONALITY_VALUE))) {
+    #warning "Invalid default personality number in RDMINIT"
```

Comment:
Warning is a compile time thing, this uses non-static data in the if. A static_cast would be more appropriate.

Reply:
Do you mean a static_cast or static_assert?

The warning correctly appeared when I'd commented out all the personalities and forgot to tweak it, I wasn't too worried at runtime if someone does something odd, more flagging up that their config is a bit wonky when working around it, rather than just silently fixing it.

Reply:
Yes, the latter. It's been quite a long day and I've been writing in every language but C++....